### PR TITLE
Riki move to target

### PR DESCRIPTION
**Pull request by omri00**

### CODE REVIEW

File: src/com/spikes2212/genericsubsystems/drivetrains/commands/targetSuppliers/NumericalTarget.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+	public NumericalTarget(Supplier<Double> position, double targetPosition, double tolerance) {
+		this.position = position;
+		this.targetPosition = targetPosition;
+		this.tolerance = tolerance;
+	}
+
+	@Override
+	public Boolean get() {
+		return targetPosition - tolerance <= position.get() && position.get() <= targetPosition + tolerance;
+	}
```
Comment (TomerSitton) [inline src/com/spikes2212/genericsubsystems/drivetrains/commands/targetSuppliers/NumericalTarget.java]: use abs. see inTargetRange in the elevator subsystem in GearworksXO repo

File: src/com/spikes2212/genericsubsystems/commands/MoveBasicSubsystemToTarget.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package com.spikes2212.genericsubsystems.commands;
+
+import java.util.function.Supplier;
+
+import com.spikes2212.genericsubsystems.BasicSubsystem;
+
+public class MoveBasicSubsystemToTarget extends MoveBasicSubsystem {
+	protected Supplier<Boolean> onTarget;
```
Comment (TomerSitton) [inline src/com/spikes2212/genericsubsystems/commands/MoveBasicSubsystemToTarget.java]: add docs

File: src/com/spikes2212/genericsubsystems/drivetrains/commands/targetSuppliers/NumericalTarget.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
+package com.spikes2212.genericsubsystems.drivetrains.commands.targetSuppliers;
+
+import java.util.function.Supplier;
+
+public class NumericalTarget implements Supplier<Boolean> {
+
```
Comment (TomerSitton) [inline src/com/spikes2212/genericsubsystems/drivetrains/commands/targetSuppliers/NumericalTarget.java]: add docs

File: src/com/spikes2212/genericsubsystems/utils/targets/NumericalTarget.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
+package com.spikes2212.genericsubsystems.utils.targets;
+
+import java.util.function.Supplier;
+
+public class NumericalTarget implements Supplier<Boolean> {
+
```
Comment (Spikes-2212) [inline src/com/spikes2212/genericsubsystems/utils/targets/NumericalTarget.java]: add docs to this class.

File: src/com/spikes2212/genericsubsystems/utils/targets/NumericalTarget.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+import java.util.function.Supplier;
+
+import com.spikes2212.genericsubsystems.BasicSubsystem;
+import com.spikes2212.genericsubsystems.commands.MoveBasicSubsystemToTarget;
+
+/**
+ * This class is a Double {@link Supplier}. It is used in the constructor of
+ * {@link MoveBasicSubsystemToTarget} to move a {@link BasicSubsystem} to a
+ * numerical target.
+ *
```
Comment (TomerSitton) [inline src/com/spikes2212/genericsubsystems/utils/targets/NumericalTarget.java]: you can give examples ("such as encoder's output and bla bla")

File: src/com/spikes2212/genericsubsystems/utils/targets/NumericalTarget.java [hunk 6]
```diff
@@ -0,0 +1,48 @@
+package com.spikes2212.genericsubsystems.utils.targets;
+
+import java.util.function.Supplier;
+
+import com.spikes2212.genericsubsystems.BasicSubsystem;
+import com.spikes2212.genericsubsystems.commands.MoveBasicSubsystemToTarget;
+
+/**
+ * This class is a Double {@link Supplier}. It is used in the constructor of
```
Comment (TomerSitton) [inline src/com/spikes2212/genericsubsystems/utils/targets/NumericalTarget.java]: double..?

File: src/com/spikes2212/genericsubsystems/commands/MoveBasicSubsystemToTarget.java [hunk 7]
```diff
@@ -0,0 +1,40 @@
+package com.spikes2212.genericsubsystems.commands;
+
+import java.util.function.Supplier;
+
+import com.spikes2212.genericsubsystems.BasicSubsystem;
+
+/**
+ * This class moves a {@link BasicSubsystem} in a given speed until reaching a
+ * target.
+ *
```
Comment (TomerSitton) [inline src/com/spikes2212/genericsubsystems/commands/MoveBasicSubsystemToTarget.java]: you can give examples in here as well. ("such as sensor or blablabla")

**Pull request merged**